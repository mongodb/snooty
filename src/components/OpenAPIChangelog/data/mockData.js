export const mockChangelog = [
  {
    date: '2024-02-01',
    paths: [
      {
        path: '/api/atlas/v2/groups/{groupId}/alertConfigs/{alertConfigId}',
        httpMethod: 'PUT',
        operationId: 'updateAlertConfiguration',
        tag: 'Alert Configurations',
        versions: [
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'removed',
            changes: [
              {
                change: 'resource version 2023-01-01 was removed, latest available resource version is 2023-02-01',
                changeCode: 'resource-version-removed',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
  {
    date: '2023-02-01',
    paths: [
      {
        path: '/api/atlas/v2/groups/byName/{groupName}',
        httpMethod: 'GET',
        operationId: 'getProjectByName',
        tag: 'Projects',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'release',
            changes: [
              {
                change: 'resource version added.',
                changeCode: 'resource-version-added',
                backwardCompatible: false,
              },
            ],
          },
        ],
      },
      {
        path: '/api/atlas/v2/groups/{groupId}/alertConfigs/{alertConfigId}',
        httpMethod: 'PUT',
        operationId: 'updateAlertConfiguration',
        tag: 'Alert Configurations',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'release',
            changes: [
              {
                change: 'resource version added.',
                changeCode: 'resource-version-added',
                backwardCompatible: false,
              },
              {
                change:
                  "for the 'query' request parameter 'clusterName', default value was changed from 'default' to 'cluster0'",
                changeCode: 'request-parameter-default-value-changed',
                backwardCompatible: false,
              },
              {
                change: "added the optional property 'links' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'deprecate',
            sunsetDate: '2024-02-01',
            changes: [
              {
                change: 'deprecated by resource version 2023-02-01 and marked for removal for 2024-02-01.',
                changeCode: 'resource-version-deprecated',
                backwardCompatible: false,
              },
            ],
          },
        ],
      },
      {
        path: '/api/atlas/v2/groups/{groupId}/accessList',
        httpMethod: 'POST',
        operationId: 'createProjectIpAccessList',
        tag: 'Project IP Access List',
        versions: [
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'update',
            changes: [
              {
                change: "added the optional property 'name' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
              {
                change: "added the optional property 'comment' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
  {
    date: '2023-02-22',
    paths: [
      {
        path: '/api/atlas/v2/groups/byName/{groupName}',
        httpMethod: 'GET',
        operationId: 'getProjectByName',
        tag: 'Projects',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'update',
            changes: [
              {
                change:
                  "added the optional property 'withDefaultAlertsSettings' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
];

export const mockDiff = [
  {
    path: '/api/atlas/v2/federationSettings/{federationSettingsId}/connectedOrgConfigs',
    httpMethod: 'GET',
    operationId: 'listConnectedOrgConfigs',
    tag: 'Federated Authentication',
    changeType: 'update',
    changes: [
      {
        change: "removed the non-success response with the status '400'.",
        changeCode: 'response-non-success-status-removed',
        backwardCompatible: false,
      },
      {
        change: "added the new optional 'query' request parameter 'envelope'.",
        changeCode: 'new-optional-request-parameter',
        backwardCompatible: true,
      },
    ],
  },
  {
    path: '/api/atlas/v2/groups/{groupId}/accessList',
    httpMethod: 'POST',
    operationId: 'createProjectIpAccessList',
    tag: 'Project IP Access List',
    changeType: 'removed',
    changes: [
      {
        change: "removed the required property 'name' from the response with the '200' status.",
        changeCode: 'response-required-property-removed',
        backwardCompatible: false,
      },
    ],
  },
];

export const mockIndex = {
  runDate: '2023-04-03',
  runId: '20230403105948111',
  specRevision: '422ab47f864909549362a7b39688404c82e2540b',
  specRevisionShort: '422ab47f864',
  versions: ['2023-01-01', '2023-02-01', '2023-08-11~preview'],
};
