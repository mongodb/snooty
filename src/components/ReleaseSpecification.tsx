import React from 'react';
import type { ReleaseSpecificationNode } from '../types/ast';
import ComponentFactory from './ComponentFactory';

interface ReleaseSpecificationProps {
  nodeData: ReleaseSpecificationNode;
}

const ReleaseSpecification = ({ nodeData: { children }, ...rest }: ReleaseSpecificationProps) => {
  return children.map((child, i) => <ComponentFactory {...rest} key={i} nodeData={child} />);
};

export default ReleaseSpecification;
