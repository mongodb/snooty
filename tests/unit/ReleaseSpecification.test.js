import React from 'react';
import { render } from '@testing-library/react';
import ReleaseSpecification from '../../src/components/ReleaseSpecification';

// data for this component
import mockData from './data/ReleaseSpecification.test.json';

it('renders correctly', () => {
  const tree = render(<ReleaseSpecification nodeData={mockData} />);
  expect(tree.asFragment()).toMatchSnapshot();
});
