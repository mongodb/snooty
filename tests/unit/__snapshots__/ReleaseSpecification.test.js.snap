// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.emotion-0 {
  border-spacing: 0;
  vertical-align: top;
  padding: 0 16px;
}

.emotion-2 {
  color: inherit;
  font-size: 13px;
  font-family: 'Source Code Pro',Menlo,monospace;
  line-height: 24px;
}

.emotion-3 {
  overflow-x: auto;
  border-radius: 0;
  border: 0;
  border-left: 2px solid;
  padding-top: 8px;
  padding-bottom: 8px;
  margin: 0;
  position: relative;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  white-space: pre;
  border-color: #E7EEEC;
  background-color: #F9FBFA;
  color: #061621;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 6px;
  padding-bottom: 6px;
}

@media only screen and (max-device-width:812px) and (-webkit-min-device-pixel-ratio:2) {
  .emotion-3 {
    white-space: pre-wrap;
  }
}

@media only screen and (min-device-width:813px) and (-webkit-min-device-pixel-ratio:2) {
  .emotion-3 {
    white-space: pre;
  }
}

.emotion-5 {
  border: 1px solid #E7EEEC;
  border-radius: 4px;
  overflow: hidden;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-1 {
  border-spacing: 0;
  width: 100%;
}

.emotion-6 {
  display: table;
  margin: 16px 0;
  min-width: 150px;
  table-layout: fixed;
  width: 100%;
}

<div
  class="emotion-6"
>
  <div
    class="emotion-5"
  >
    <div
      class="emotion-4"
    >
      <pre
        class="emotion-3"
        tabindex="-1"
      >
        <code
          class="lg-highlight-hljs-light emotion-2 none"
        >
          <table
            class="emotion-1"
          >
            <tbody>
              <tr
                class=""
              >
                <td
                  class="emotion-0"
                >
                  test code
                </td>
              </tr>
            </tbody>
          </table>
        </code>
      </pre>
    </div>
  </div>
</div>
`;
